src/views/datasources/list/CreateDataSourceModal/CreateDataSourceModal.tsx

```diff
@@ -0,0 +1,232 @@
+import {
+  Form,
+  FormGroup,
+  NumberInput,
+  Popover,
+  Stack,
+  StackItem,
+  TextInput,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import { CRON_DOC_URL } from '../../dataimportcron/details/DataImportCronManageModal/utils';
+
+import { createDataSourceWithImportCron, generateDataSourceName } from './utils';
+
+export type CreateDataSourceModalFormType = {
+  name: string;
+  size: string;
+  url: string;
+  importsToKeep: number;
+  schedule: string;
+};
+
+type CreateDataSourceModalProps = {
+  isOpen: boolean;
+  namespace: string;
+  onClose: () => void;
+};
+
+export const CreateDataSourceModal: React.FC<CreateDataSourceModalProps> = ({
+  isOpen,
+  namespace,
+  onClose,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const history = useHistory();
+
+  const {
+    register,
+    formState: { errors },
+    setValue,
+    watch,
+    handleSubmit,
+  } = useForm<CreateDataSourceModalFormType>({
+    defaultValues: {
+      name: generateDataSourceName(),
+      size: '30Gi',
+      importsToKeep: 3,
+    },
+  });
+  const importsToKeep = watch('importsToKeep');
+  const size = watch('size');
+  const name = watch('name');
+
+  const onSubmit = handleSubmit(
+    (data) =>
+      createDataSourceWithImportCron({
+        ...data,
+        namespace: namespace || 'default',
+      }).then(() =>
+        history.push(`/k8s/ns/${namespace || 'default'}/${DataSourceModelRef}/${name}`),
+      ),
+    () => Promise.reject(t('Missing required fields')),
+  );
+
+  return (
+    <TabModal
+      isOpen={isOpen}
+      onClose={onClose}
+      onSubmit={() => onSubmit()}
+      headerText={t('Create DataSource')}
+    >
+      <Stack hasGutter>
+        <StackItem>
+          <MutedTextSpan
+            text={t(
+              'A DataImportCron will also be created, which will define a cron job for recurring polling/importing of the disk image',
+            )}
+          />
+        </StackItem>
+        <StackItem>
+          <Form>
+            <FormGroup
+              fieldId="datasource-create-name"
+              label={t('Name')}
+              validated={errors?.['name'] ? ValidatedOptions.error : ValidatedOptions.default}
+              helperTextInvalid={t('This field is required')}
+              helperTextInvalidIcon={<RedExclamationCircleIcon title="Error" />}
+              isRequired
+            >
+              <FormTextInput
+                {...register('name', { required: true })}
+                id="datasource-create-name"
+                type="text"
+                aria-label={t('Name')}
+                data-test-id="datasource-create-name"
+                validated={errors?.['name'] ? ValidatedOptions.error : ValidatedOptions.default}
+              />
+            </FormGroup>
+            <FormGroup fieldId="datasource-create-source-type" label={t('Source type')}>
```

Comment:
this shouldn't have any validation? for non empty or something?

Reply:
No need because right now this field is disabled, only Registry is allowed